atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/SweepMetricsTest.java

```diff
@@ -0,0 +1,135 @@
+import com.codahale.metrics.ConsoleReporter;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.primitives.Longs;
+import com.palantir.atlasdb.keyvalue.api.SweepResults;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.util.AtlasDbMetrics;
+
+public class SweepMetricsTest {
+    private static final TableReference TABLE = TableReference.createFromFullyQualifiedName("test.table");
+    private static final TableReference OTHER_TABLE = TableReference.createFromFullyQualifiedName("test.other_table");
+
+    private static final long DELETED = 10L;
+    private static final long EXAMINED = 15L;
+    private static final SweepResults SWEEP_RESULTS_FOR_TABLE = SweepResults.builder()
+            .cellsDeleted(DELETED)
+            .cellsExamined(EXAMINED)
+            .sweptTimestamp(1337L)
+            .build();
+
+    private static final long OTHER_DELETED = 12L;
+    private static final long OTHER_EXAMINED = 4L;
+    private static final SweepResults SWEEP_RESULTS_FOR_OTHER_TABLE = SweepResults.builder()
+            .cellsDeleted(OTHER_DELETED)
+            .cellsExamined(OTHER_EXAMINED)
+            .sweptTimestamp(1338L)
+            .build();
+
+    private static final MetricRegistry METRIC_REGISTRY = AtlasDbMetrics.getMetricRegistry();
+
+    private SweepMetrics sweepMetrics;
+
+    @Before
+    public void setUp() {
+        sweepMetrics = new SweepMetrics();
+        sweepMetrics.registerMetricsIfNecessary(TABLE);
+    }
+
+    @After
+    public void tearDown() {
+        METRIC_REGISTRY.removeMatching((name, metric) -> true);
+    }
+
+    @Test
+    public void cellsDeletedAreRecorded() {
+        sweepMetrics.recordMetrics(TABLE, SWEEP_RESULTS_FOR_TABLE);
+
+        assertCellsDeleted(TABLE, DELETED);
+    }
+
+    @Test
+    public void cellsDeletedAreRecordedSeparatelyAndAggregated() {
+        sweepMetrics.recordMetrics(TABLE, SWEEP_RESULTS_FOR_TABLE);
+        sweepMetrics.recordMetrics(OTHER_TABLE, SWEEP_RESULTS_FOR_OTHER_TABLE);
+
+        assertCellsDeleted(TABLE, DELETED);
+        assertCellsDeleted(OTHER_TABLE, OTHER_DELETED);
+        assertValuesRecorded(SweepMetrics.STALE_VALUES_DELETED, DELETED, OTHER_DELETED);
+    }
+
+    @Test
+    public void cellsExaminedAreRecorded() {
+        sweepMetrics.recordMetrics(TABLE, SWEEP_RESULTS_FOR_TABLE);
+
+        assertCellsExamined(TABLE, EXAMINED);
+    }
+
+    @Test
+    public void cellsExaminedAreRecordedSeparatelyAndAggregated() {
+        sweepMetrics.recordMetrics(TABLE, SWEEP_RESULTS_FOR_TABLE);
+        sweepMetrics.recordMetrics(OTHER_TABLE, SWEEP_RESULTS_FOR_OTHER_TABLE);
+
+        assertCellsExamined(TABLE, EXAMINED);
+        assertCellsExamined(OTHER_TABLE, OTHER_EXAMINED);
+        assertValuesRecorded(SweepMetrics.CELLS_EXAMINED, EXAMINED, OTHER_EXAMINED);
+    }
+
+    @Ignore // This is just for me to run locally and check out what the metrics reports look like
```

Comment:
We probably want to remove this eventually, but may be useful for future metrics?